### PR TITLE
More coverage

### PR DESCRIPTION
**Pull request by Thrameos**
Yet another coverage pull request.

### CODE REVIEW

File: native/common/jp_field.cpp [hunk 1]
```diff
@@ -49,12 +49,6 @@ void JPField::setStaticField(PyObject *pyobj)
 {
 	JP_TRACE_IN("JPField::setStaticAttribute");
 	JPJavaFrame frame(m_Class->getContext());
-	if (isFinal())
```
Comment (marscher) [inline native/common/jp_field.cpp]: This condition could never happen?

File: test/jpypetest/test_jobject.py [hunk 2]
```diff
@@ -166,3 +167,34 @@ def testObjectDouble(self):
...
+    def testObjectField(self):
+        with self.assertRaises(TypeError):
+            self.fixture.object_field = object()
+        with self.assertRaises(TypeError):
+            self.fixture.static_object_field = object()
+
+    def testArraySetRangeFail(self):
+        ja = JArray(JObject)(4)
+        with self.assertRaises(TypeError):
+            ja[:] = [ 1, 2, object(), 3]
```
Comment (marscher) [inline test/jpypetest/test_jobject.py]: pep8 pls (extra space in beginning of list definition)

File: native/common/jp_field.cpp (hunk 1 above)
Comment (Thrameos) [reply native/common/jp_field.cpp]: This was a problem with responsibilities.  This condition was never hit  and in tracing I discovered that the PyJPField had already checked against this condition.  Thus this check was redundant.

File: test/jpypetest/test_jobject.py (hunk 2 above)
Comment (Thrameos) [reply test/jpypetest/test_jobject.py]: No problem.  I will pep8 prior to merge.

File: native/common/jp_field.cpp (hunk 1 above)
Comment (marscher) [reply native/common/jp_field.cpp]: Thanks for the explanation.

**Pull request merged**